ql/src/test/queries/clientpositive/test_unixtimestamp_for_invaild_dates.q

```diff
@@ -0,0 +1,20 @@
+DESCRIBE FUNCTION UNIX_TIMESTAMP;
```

Comment:
Please add or point to tests (preferably unit tests) with unix_timestamp, invalid dates, and no pattern. 

For instance:
```
unix_timestamp('2001-02-31')
```
Is the behavior of the function inline with the variant that accepts a pattern?

Reply:
Sure , i have added the patterns in `TestGenericUDFToUnixTimestampEvaluateStringStringCsv` .
Also i will remove this q file which is now redundant because all the cases are covered in the csv.

Reply:
The `TestGenericUDFToUnixTimestampEvaluateStringStringCsv` only tests the UDF using two inputs (date + pattern). I was wondering if we have or if we should add tests (with invalid dates) where the UDF has only one input (date and no pattern).

I would like to see if there are inconsistencies among the following
```
unix_timestamp('2001-02-31')
unix_timestamp('2001-02-31', 'uuuu-MM-dd')
```
after the changes in this PR.

Reply:
Tested Without this PR, I see that : 
1. without the pattern : UDF returns Null
2. With pattern
For SIMPLE:
> u ---> Day number of week
> y ---> Year

For DateTime:
> u ---> year
> y ---> year-of-ear

```
7: jdbc:hive2://localhost:10001/> set hive.datetime.formatter=SIMPLE;
No rows affected (0.018 seconds)

7: jdbc:hive2://localhost:10001/> select to_unix_timestamp('2001-02-31') ;
+-------+
|  _c0  |
+-------+
| NULL  |
+-------+
1 row selected (0.125 seconds)

7: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'uuuu-MM-dd');
+----------+
|   _c0    |
+----------+
| 5270400  |
+----------+
1 row selected (0.147 seconds)

7: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'yyyy-MM-dd');
+------------+
|    _c0     |
+------------+
| 983577600  |   this shows timestamp of March 3, 2001 12AM GMT
+------------+
1 row selected (0.131 seconds)


7: jdbc:hive2://localhost:10001/> set hive.datetime.formatter=DATETIME;
No rows affected (0.023 seconds)



7: jdbc:hive2://localhost:10001/> select to_unix_timestamp('2001-02-31') ;
+-------+
|  _c0  |
+-------+
| NULL  |
+-------+

7: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'uuuu-MM-dd');

+------------+
|    _c0     |
+------------+
| 983318400  |  This shows time stamp of February 28, 2001 12AM GMT
+------------+
```

Reply:
With this PR: 
a) without the pattern : UDF returns Null  ->same as before
b) With pattern
1. For SIMPLE and SMART:  Same behavior as before this fix
2. For DATETIME and SMART: Same behavior as before this fix
3. For DATETIME and STRICT: Invalid dates are returned as null
4. For DATETIME and LENIENT : Matches the output of SIMPLE  (but the symbol uuuu represents year instead of Day number of week)

```
8: jdbc:hive2://localhost:10001/> set hive.datetime.formatter=SIMPLE;
No rows affected (0.048 seconds)

8: jdbc:hive2://localhost:10001/> set hive.datetime.resolver.style=SMART;
8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31');
No rows affected (0.018 seconds)
+-------+
|  _c0  |
+-------+
| NULL  |
+-------+
1 row selected (0.21 seconds)

8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'uuuu-MM-dd');
+----------+
|   _c0    |
+----------+
| 5270400  |
+----------+
1 row selected (0.141 seconds)
8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'yyyy-MM-dd');
+------------+
|    _c0     |
+------------+
| 983577600  |
+------------+
1 row selected (0.135 seconds)


8: jdbc:hive2://localhost:10001/> set hive.datetime.formatter=DATETIME;
No rows affected (0.011 seconds)
8: jdbc:hive2://localhost:10001/> set hive.datetime.resolver.style=SMART;
No rows affected (0.016 seconds)
8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31');
+-------+
|  _c0  |
+-------+
| NULL  |
+-------+
1 row selected (0.195 seconds)
8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'uuuu-MM-dd');
+------------+
|    _c0     |
+------------+
| 983318400  |
+------------+
1 row selected (0.123 seconds)


8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'yyyy-MM-dd');
+------------+
|    _c0     |
+------------+
| 983318400  |
+------------+
1 row selected (0.132 seconds)


8: jdbc:hive2://localhost:10001/> set hive.datetime.resolver.style=STRICT;
No rows affected (0.019 seconds)

8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31');
+-------+
|  _c0  |
+-------+
| NULL  |
+-------+
1 row selected (0.122 seconds)

8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'uuuu-MM-dd');
+-------+
|  _c0  |
+-------+
| NULL  |
+-------+
1 row selected (0.277 seconds)

8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'yyyy-MM-dd');
+-------+
|  _c0  |
+-------+
| NULL  |
+-------+
1 row selected (0.125 seconds)


8: jdbc:hive2://localhost:10001/> set hive.datetime.resolver.style=LENIENT;
No rows affected (0.024 seconds)

8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'yyyy-MM-dd');
+------------+
|    _c0     |
+------------+
| 983577600  |
+------------+
1 row selected (0.124 seconds)

8: jdbc:hive2://localhost:10001/> select unix_timestamp('2001-02-31', 'uuuu-MM-dd');
+------------+
|    _c0     |
+------------+
| 983577600  |
+------------+
1 row selected (0.101 seconds)
8:
```

Reply:
There is a small inconsistency in the fact that the new property controlling the resolution style does not take effect when the there is no pattern argument.

```sql
set hive.datetime.formatter=DATETIME;
set hive.datetime.formatter.resolver.style=STRICT
set hive.datetime.formatter.resolver.style=SMART
set hive.datetime.formatter.resolver.style=LENIENT
select unix_timestamp('2001-02-31');
NULL
```
No matter the value of the resolver the result will always be `null` so this could be considered a bug by some people. This is not a regression though, so I guess we could deal with it in a follow-up ticket if necessary.